framework/include/utils/PetscSupport.h

```diff
@@ -153,6 +196,15 @@ void setSinglePetscOption(const std::string & name,
                           const std::string & value = "",
                           FEProblemBase * const problem = nullptr);
 
+/**
+Same as setSinglePetscOption, but does not set the option if it belongs to the given list of
+ignored options.
+*/
+void setSinglePetscOptionIfNotIgnored(const MultiMooseEnum & ignored_options,
```

Comment:
```suggestion
void setSinglePetscOptionIfAppropriate(const MultiMooseEnum & ignored_options,
```
or something like that